tests/main/upgrade-from-release/task.yaml

```diff
@@ -62,7 +62,13 @@ execute: |
     snap install go-example-webserver
     tests.systemd wait-for-service -n 30 --state active snap.go-example-webserver.webserver.service
     snap install test-snapd-tools
-    test-snapd-tools.echo hello | MATCH hello
+    # In case of known error on older snapd deb versions e.g. 2.32.5+18.04,
```

Comment:
This is the problematic behavior: 
```
root@bionic-container:~# test-snapd-tools.echo hello
hello
root@bionic-container:~# SNAP_REEXEC=0 test-snapd-tools.echo hello
cannot perform readlinkat() on the mount namespace file descriptor of the init process: Permission denied
```

Reply:
This scenario is not exercised as tests are run in this PR, but was specifically tested locally with SPREAD_SNAPD_DEB_FROM_REPO=false SPREAD_MODIFY_CORE_SNAP_FOR_REEXEC=0 SPREAD_TRUST_TEST_KEYS=false SPREAD_SNAP_REEXEC=0 SPREAD_CORE_CHANNEL=stable.

Reply:
I don't really understand why it doesn't fail in our ci if the error happens before we install the current deb, so we are using the deb from main which is the same in all the contexts.

Reply:
Presume it re-executes to either core or built snapd snap?

Reply:
The snaps it installs requires core snap, so that gets installed and by default it re-executes snapd from core.